### PR TITLE
python-box 7.1.1

### PR DESCRIPTION
**Pull request by psteyer**
python-box 7.1.1 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4825](https://anaconda.atlassian.net/browse/PKG-4825) 
- [Upstream repository](https://github.com/cdgriffith/Box/tree/7.1.1)
- [Upstream changelog/diff](https://github.com/cdgriffith/Box/compare/5.4.1...7.1.1)

### Explanation of changes:

- changing source of download for tests
- removed noarch
- adding cython
- adding run_constrained
- adding test file
- running pytest lines

[PKG-4825]: https://anaconda.atlassian.net/browse/PKG-4825?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,37 +1,52 @@
...
+    - wheel
+    - setuptools
   run:
-    - python >=3.6
-    - pip
-    - ruamel.yaml
-    - toml
+    - python
+  run_constrained:
+    - msgpack >=1.0.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - msgpack-python >=1.0.0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,37 +1,52 @@
...
     - box
   commands:
     - pip check
+    # skipping test due to not having msgpack. Skipping other tests that require it.
+    - pytest -v test --ignore=test/test_converters.py -k "not (msgpack or test_from_all)" # [not s390x]
+    - pytest -v test --ignore=test/test_converters.py -k "not (msgpack or test_from_all or test_to_yaml_file)" # [s390x]
+  requires:
+    - pip
+    - pytest >=7.1.3
+    - ruamel.yaml >=0.17
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:48]:
```suggestion
    - ruamel.yaml >=0.17
    - msgpack-python >=1.0.0
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,37 +1,52 @@
...
 test:
+  source_files:
+    - test
   imports:
     - box
   commands:
     - pip check
+    # skipping test due to not having msgpack. Skipping other tests that require it.
+    - pytest -v test --ignore=test/test_converters.py -k "not (msgpack or test_from_all)" # [not s390x]
+    - pytest -v test --ignore=test/test_converters.py -k "not (msgpack or test_from_all or test_to_yaml_file)" # [s390x]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Could you try with `msgpack-python`?

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,37 +1,53 @@
...
 test:
+  source_files:
+    - test
   imports:
     - box
   commands:
     - pip check
+    # skipping test due to not having msgpack. Skipping other tests that require it.
+    - pytest -v test --ignore=test/test_converters.py -k "not test_from_all" # [not s390x]
+    - pytest -v test --ignore=test/test_converters.py -k "not (test_from_all or test_to_yaml_file)" # [s390x]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Is it still relevant with `msgpack-python >=1.0.0`?

**Review by skupr-anaconda — APPROVED** [review]
LGTM but skipped test

File: recipe/meta.yaml (hunk 4 above)
Comment (psteyer) [reply recipe/meta.yaml]: I can check!

Comment (psteyer) [reply recipe/meta.yaml]: @skupr-anaconda 
Looks like for `s390x`, the test `TestBox.test_to_yaml_file` still fails with a `_ruamel_yaml.pyx:904: ReaderError `

Comment (skupr-anaconda) [reply recipe/meta.yaml]: you can skip that test on s390x

**Pull request merged**